build/scripts/check-toolset-insertion.ps1

```diff
@@ -0,0 +1,126 @@
+Param(
+    [string]$sourcePath = $null,
+    [string]$binariesPath = $null
+)
+set-strictmode -version 2.0
+$ErrorActionPreference="Stop"
+
+[Void][Reflection.Assembly]::LoadWithPartialName('System.IO.Compression.FileSystem')
```

Comment:
> My understanding is that powershell will exit with 1 in case of exceptions?

Sadly this is not the case.  It's definitely how you'd like a scripting language to act but it's not how Powershell works.